sklearn/model_selection/tests/test_search.py

```diff
@@ -198,6 +198,21 @@ def test_grid_search():
     assert_raises(ValueError, grid_search.fit, X, y)
 
 
+def test_grid_search_pipeline_steps():
+    pipe = Pipeline([('regressor', LinearRegression())])
+    param_grid = {'regressor': [LinearRegression(), Ridge()]}
+    grid_search = GridSearchCV(pipe, param_grid, cv=2)
+    grid_search.fit(X, y)
+    regressor_results = grid_search.cv_results_['param_regressor']
+    assert isinstance(regressor_results[0], LinearRegression)
+    assert isinstance(regressor_results[1], Ridge)
+    assert not hasattr(regressor_results[0], 'coef_')
+    assert not hasattr(regressor_results[1], 'coef_')
+    # check that we didn't modify the parameter grid that was passed
+    assert not hasattr(param_grid['regressor'][0], 'coef_')
+    assert not hasattr(param_grid['regressor'][1], 'coef_')
```

Comment:
`assert regressor_results[i] is not grid_search.best_estimator_` ?

Reply:
well they are not fitted and ``best_estimator_`` is fitted. but I can add another check?

Reply:
yeah, doesn't hurt, and it specifically addresses https://github.com/scikit-learn/scikit-learn/issues/10063#issuecomment-535591894

Reply:
done